### ISSUE TITLE
An error in API Reference doc for pytorch_lightning.callbacks.BaseFinetuning.filter_params

### ISSUE THREAD

**Issue by xiaoyuan0203** [Lightning-AI/lightning#13238]
## 📚 Documentation

The API Reference doc for `pytorch_lightning.callbacks.BaseFinetuning.filter_params` says that: 
the `train_bn` parameter notes whether to train BatchNorm module.

But the source code implements as below:

```python
@staticmethod
def filter_params(
    modules: Union[Module, Iterable[Union[Module, Iterable]]], train_bn: bool = True, requires_grad: bool = True
) -> Generator:
    modules = BaseFinetuning.flatten_modules(modules)
    for mod in modules:
        if isinstance(mod, _BatchNorm) and not train_bn:
            continue
        # recursion could yield duplicate parameters for parent modules w/ parameters so disabling it
        for param in mod.parameters(recurse=False):
            if param.requires_grad == requires_grad:
                yield param
```

Whether the parametes `requires_grad` set to `True` or `False`, the BatchNorm module in the `modules` will be ignored, so the parameter `train_bn` should be described as this: Whether not to ignore the BatchNorm module.

cc @borda @rohitgr7

**Comment by xiaoyuan0203** [comment]
If I want to change the doc, where to do? The source docstring?

**Comment by akihironitta** [comment]
It's in the docstring in the source file.
https://github.com/Lightning-AI/lightning/blob/cf189bdc9b32326ea02b9af6dd6e1bf04f0e4d2e/src/pytorch_lightning/callbacks/finetuning.py

**Comment by xiaoyuan0203** [comment]
@akihironitta  Can I open a PR and merge into master branch while what you provide is not a master branch?

**Comment by akihironitta** [comment]
> @akihironitta  Can I open a PR and merge into master branch ...?

Yes! That'll be fantastic!

> while what you provide is not a master branch

It *was* `master` at the time of writing the comment! I always paste links with a specific commit hash so that anyone who will look back doesn't see 404 (due to a changed filename etc.) or the wrong (i.e., outdated) line number :)

**Comment by nishantb06** [comment]
@xiaoyuan0203 if you're not working on this issue, can i pick it up?

**Comment by xiaoyuan0203** [comment]
OK
